File: python/anagram/anagram.py
from collections import Counter
from functools import partial

def is_anagram(word, candidate):
    word = word.lower()
    candidate = candidate.lower()
    return Counter(word) == Counter(candidate) and word != candidate

def detect_anagrams(word, candidates):
    return [candidate
            for candidate in candidates
            if is_anagram(word, candidate)]

File: haskell/anagram/Anagram.hs
module Anagram (anagramsFor) where
import Data.List(sort)
import Data.Char(toLower)

anagramsFor :: String -> [String] -> [String]
anagramsFor word candidates = filter match candidates
    where
        match candidate = loweredWord /= lower candidate && sortedWord == sort (lower candidate)
        lower = map toLower 
        loweredWord = lower word
        sortedWord = sort loweredWord
